packages/features/ee/teams/components/MemberListItem.tsx

```diff
@@ -353,26 +602,124 @@ export default function MemberListItem(props: Props) {
         </Dialog>
       )}
 
-      {showChangeMemberRoleModal && (
+      {state.changeMemberRole.showModal && (
         <MemberChangeRoleModal
-          isOpen={showChangeMemberRoleModal}
+          isOpen={true}
           currentMember={props.team.membership.role}
           teamId={props.team?.id}
-          memberId={props.member.id}
-          initialRole={props.member.role as MembershipRole}
-          onExit={() => setShowChangeMemberRoleModal(false)}
+          memberId={state.changeMemberRole.user?.id as number}
+          initialRole={state.changeMemberRole.user?.role as MembershipRole}
+          onExit={() =>
+            dispatch({
+              type: "CLOSE_MODAL",
+            })
+          }
+        />
+      )}
+      {state.inviteMember.showModal && (
+        <MemberInvitationModal
+          isPending={inviteMemberMutation.isPending}
+          isOpen={true}
+          orgMembers={props.orgMembersNotInThisTeam}
+          members={props.members}
```

Comment:
fetching all the members is not scalable

Reply:
Can you raise a follow up PR after this is merged https://github.com/calcom/cal.com/pull/16155 and use `MemberInvitationModalWithoutMembers`?

Reply:
Let's wait until your PR is merged. I have one more that depends on this PR being merged. In the meantime, I will fix the other things.